lms/views/api/grant_token.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+from urllib.parse import urlparse
+
+import jwt
+from pyramid.view import view_config
+
+
+@view_config(permission="api", renderer="json", route_name="api.grant_token")
+def grant_token(request):
+    """
+    Return a grant token that the Hypothesis client can use to log in to H.
+    """
```

Comment:
The code in this function is largely copied and tweaked from `JSConfig._grant_token`. Is there a good shared place to put this logic?

Reply:
What is the `context` object during one of these `/api/grant_token` requests?

For example during an LTI launch request the context is an [`LTILaunchResource`](https://github.com/hypothesis/lms/blob/master/lms/resources/lti_launch.py). There's also [`OAuth2RedirectResource`](https://github.com/hypothesis/lms/blob/master/lms/resources/oauth2_redirect.py) for OAuth 2 redirect requests.

I think you may want to add a `GrantTokenResource` class for this `/grant_token` route in `lms/resources/`. That resource class can then have a `context.grant_token` property that would call the code that's currently in `JSConfig._grant_token()` (that code would move to a private helper function within `/lms/resources/`, called by both `JSConfig` and the new resource class).